velox/core/PlanNode.h

```diff
@@ -573,6 +573,19 @@ class AggregationNode : public PlanNode {
     return preGroupedKeys_;
   }
 
+  bool isPreGrouped() const {
+    return !preGroupedKeys_.empty() &&
+        std::equal(
```

Comment:
This does not semantically equal to `aggregationNode->preGroupedKeys().size() == aggregationNode->groupingKeys().size()`, for example if they have different orders, previous returns true but this one returns false. Not sure if this can happen. cc @ulysses-you @mbasmanova

Reply:
@waitinfuture This is a good point. preGroupedKeys should be a subset of groupingKeys, but there is not requirements for the order of keys in preGroupedKeys to match the order of keys in groupingKeys.

Reply:
yeah, it depends on how to develop with this interface. If the developer knows the input data are pre-grouped, then just set all grouping keys as preGroupedKeys (that's what Gluten did).  Before, it is a bit flaky by checking grouping key size, e.g.,  developer may set atrribute which is not a grouping key.

Reply:
I think the safe way is to check every key in `preGroupedKeys_` is in `groupingKeys_`, but not necessarily with the same position.

Reply:
Perhaps, a better way is to require that preGroupedKeys is a subset of groupingKeys and keys appear in the same order. We can add a check to the constructor to enforce that.